src/docserv/functions.py

```diff
@@ -32,6 +33,49 @@ def mail(text, subject, to):
     p.communicate(msg.as_string())
 
 
+def get_all_file_paths(directory, zip_formats):
+    """Collect all files with paths in specified formats from given directory.
+
+    :param str directory: a directory from which files are to be collected
+    :param list zip_formats: a list containing accepted documentation formats
+        example: ["pdf", "epub", "single-html"]
+    :return: a list with all files matching specified documentation formats
+    :rtype: list
+    """
+    file_paths = []
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            format_ok = 0
+            for format in zip_formats:
+                if format in filepath:
+                    format_ok = 1
+                elif root is directory and format in filepath:
+                    root_ok = 1
+            if format_ok:
+                file_paths.append(filepath)
+    return file_paths
+
+def create_zip_file(directory, zip_formats, zip_file):
+    """Create zip archive with all files in specified formats from given directory.
+
+    :param str directory: a directory/folder from which files are to be collected
+    :param list zip_formats: a list containing accepted documentation formats
+        example: ["pdf", "epub", "single-html"]
+    :param str zip_file:  archive name including path where to be saved
+    :return: a list with all files matching specified documentation formats
+    :rtype: list
+    """
+    # calling function to get all file paths in the directory
+    file_paths = get_all_file_paths(directory, zip_formats)
+    # writing files to a zipfile
+    with ZipFile(zip_file, 'w', ZIP_DEFLATED) as zip:
+        # writing each file one by one
+        for filepath in file_paths:
```

Comment:
... and add it here:

```suggestion
        for filepath in get_all_file_paths(directory, zip_formats):
```